### PR TITLE
internal: Remove commented out debug fmt.Prints.

### LINKED ISSUE #53: internal: remove commented-out debugging block from production generated code
See https://github.com/goccmack/gocc/pull/52#discussion_r129208442

### PR DESCRIPTION
**Pull request by sangisos**
Removes commented out prints for debug purposes from production code and removes
depricated debug printouts generated from lexer.go and parser.go.

Also fixes minor issues in template code for gofmt complience.

Fixes #53.

### CODE REVIEW

**Comment by sangisos** [comment]
Some benchmarking on https://github.com/llir/llvm:
```
gocc -zip ll.bnf

old lexer run 1
make  243.09s user 1.10s system 113% cpu 3:34.29 total

old lexer run 2
make  242.63s user 1.12s system 114% cpu 3:32.59 total

old lexer run 3
make  246.06s user 1.00s system 114% cpu 3:36.19 total

old lexer run 4
make  245.41s user 1.09s system 113% cpu 3:36.72 total

old lexer run 5
make 243.37s user 1.09s system 112% cpu 3:37.21 total


new lexer run 1
make  239.61s user 1.10s system 112% cpu 3:34.20 total

new lexer run 2
make  242.72s user 1.11s system 114% cpu 3:33.28 total

new lexer run 3
make  240.61s user 1.09s system 112% cpu 3:34.67 total

new lexer run 4
make &> /dev/null  245.67s user 1.04s system 114% cpu 3:34.66 total

new lexer run 5
make &> /dev/null  242.82s user 1.09s system 112% cpu 3:35.94 total
```

**Comment by mewmew** [comment]
tl;dr: LGTM

The code looks good to me. The run time of the code generation (i.e. generation of Go source code by gocc; `gocc -zip ll.bnf`) should not be affected by this PR (or only moderately affected). However, the run time of the generated lexer (and parser) might be.

The rationale for this is that the template code of the code generator is run exactly once per Go source code being generated, while the generated code is executed several times to tokenize the input source files, and perform shift/reductions on the stream of tokens. Both of these tasks may take substantial amount of time as they are invoked repeatedly (and thus small changes in run time, may accumulate to large differences in total run time).

For the parser, the github.com/llir/llvm/asm/internal/parser benchmark was used, and for the lexer the github.com/mewmew/uc/gocc/lexer benchmark was used.

# Benchmarks from master branch

## Lexer benchmarks

```
u@x220 ~/D/g/s/g/m/u/g/parser> go test -bench=. github.com/mewmew/uc/gocc/lexer
BenchmarkLexer-4   	   10000	    200949 ns/op	  15.29 MB/s
PASS
ok  	github.com/mewmew/uc/gocc/lexer	2.035s
u@x220 ~/D/g/s/g/m/u/g/parser> go test -bench=. github.com/mewmew/uc/gocc/lexer
BenchmarkLexer-4   	   10000	    209528 ns/op	  14.66 MB/s
PASS
ok  	github.com/mewmew/uc/gocc/lexer	2.118s
u@x220 ~/D/g/s/g/m/u/g/parser> go test -bench=. github.com/mewmew/uc/gocc/lexer
BenchmarkLexer-4   	   10000	    205191 ns/op	  14.97 MB/s
PASS
ok  	github.com/mewmew/uc/gocc/lexer	2.079s
u@x220 ~/D/g/s/g/m/u/g/parser> go test -bench=. github.com/mewmew/uc/gocc/lexer
BenchmarkLexer-4   	   10000	    203496 ns/op	  15.10 MB/s
PASS
ok  	github.com/mewmew/uc/gocc/lexer	2.062s
u@x220 ~/D/g/s/g/m/u/g/parser> go test -bench=. github.com/mewmew/uc/gocc/lexer
BenchmarkLexer-4   	   10000	    204611 ns/op	  15.01 MB/s
PASS
ok  	github.com/mewmew/uc/gocc/lexer	2.076s
```

## Parser benchmarks

```
u@x220 ~/D/g/s/g/l/llvm> go test -bench=. github.com/llir/llvm/asm/internal/parser
BenchmarkParseFile-4   	       1	12443466679 ns/op
PASS
ok  	github.com/llir/llvm/asm/internal/parser	12.850s

u@x220 ~/D/g/s/g/l/llvm> go test -bench=. github.com/llir/llvm/asm/internal/parser
BenchmarkParseFile-4   	       1	12522431079 ns/op
PASS
ok  	github.com/llir/llvm/asm/internal/parser	12.996s

u@x220 ~/D/g/s/g/l/llvm> go test -bench=. github.com/llir/llvm/asm/internal/parser
BenchmarkParseFile-4   	       1	12506326553 ns/op
PASS
ok  	github.com/llir/llvm/asm/internal/parser	12.966s
```

# Benchmarks from sangisos/remcom branch

## Lexer benchmarks

```
u@x220 ~/D/g/s/g/m/u/g/lexer> go test -bench=. github.com/mewmew/uc/gocc/lexer
BenchmarkLexer-4   	    5000	    205619 ns/op	  14.94 MB/s
PASS
ok  	github.com/mewmew/uc/gocc/lexer	1.062s
u@x220 ~/D/g/s/g/m/u/g/lexer> go test -bench=. github.com/mewmew/uc/gocc/lexer
BenchmarkLexer-4   	    5000	    202892 ns/op	  15.14 MB/s
PASS
ok  	github.com/mewmew/uc/gocc/lexer	1.047s
u@x220 ~/D/g/s/g/m/u/g/lexer> go test -bench=. github.com/mewmew/uc/gocc/lexer
BenchmarkLexer-4   	   10000	    207512 ns/op	  14.80 MB/s
PASS
ok  	github.com/mewmew/uc/gocc/lexer	2.098s
u@x220 ~/D/g/s/g/m/u/g/lexer> go test -bench=. github.com/mewmew/uc/gocc/lexer
BenchmarkLexer-4   	   10000	    206367 ns/op	  14.89 MB/s
PASS
ok  	github.com/mewmew/uc/gocc/lexer	2.089s
u@x220 ~/D/g/s/g/m/u/g/lexer> go test -bench=. github.com/mewmew/uc/gocc/lexer
BenchmarkLexer-4   	    5000	    203743 ns/op	  15.08 MB/s
PASS
ok  	github.com/mewmew/uc/gocc/lexer	1.050s
```

## Parser benchmarks

```
u@x220 ~/D/g/s/g/l/llvm> go test -bench=. github.com/llir/llvm/asm/internal/parser
BenchmarkParseFile-4   	       1	11917399834 ns/op
PASS
ok  	github.com/llir/llvm/asm/internal/parser	12.368s

u@x220 ~/D/g/s/g/l/llvm> go test -bench=. github.com/llir/llvm/asm/internal/parser
BenchmarkParseFile-4   	       1	12118601151 ns/op
PASS
ok  	github.com/llir/llvm/asm/internal/parser	12.556s
u@x220 ~/D/g/s/g/l/llvm> go test -bench=. github.com/llir/llvm/asm/internal/parser
BenchmarkParseFile-4   	       1	11955297265 ns/op
PASS
ok  	github.com/llir/llvm/asm/internal/parser	12.468s
```

Based on these benchmarks, I see no reason to hold back this PR; especially since it solves a real issue, that of introducing heisenbugs that are impossible to debug (when the debug code differs from that in production).

**Comment by mewmew** [comment]
@awalterschulze Feel free to merge if you agree with my LGTM.

**Comment by sangisos** [comment]
@mewmew yes. That makes sense. This is why I wanted you to make the bench =).  Thanks.

**Comment by mewmew** [comment]
> @mewmew yes. That makes sense. This is why I wanted you to make the bench =). Thanks.

:)

Must say, I'm a bit jealous of your contrib stats for this PR, any diff removing more lines than adding is a good diff in my book.

```diff
+40
-257
```

**Comment by awalterschulze** [comment]
Nice work! Both of you :)
